clients/banking/src/components/BeneficiaryList.tsx

```diff
@@ -62,47 +89,89 @@ const Cell = (props: BoxProps) => (
   />
 );
 
-const smallColumns: ColumnConfig<GetNode<Beneficiaries>, undefined>[] = [
+export const getBeneficiaryIdentifier = (beneficiary: Beneficiary) =>
+  match(beneficiary)
+    .returnType<{ label: string; text: string }>()
+    .with({ __typename: "TrustedInternalBeneficiary" }, ({ accountId }) => ({
+      label: t("beneficiaries.accountIdentifier.accountId"),
+      text: accountId,
+    }))
+    .with({ __typename: "TrustedSepaBeneficiary" }, ({ iban }) => ({
+      label: t("beneficiaries.accountIdentifier.iban"),
+      text: printFormat(iban),
+    }))
+    .with({ __typename: "TrustedInternationalBeneficiary" }, ({ details }) =>
+      match(Object.fromEntries(details.map(({ key, value }): [string, string] => [key, value])))
+        .with({ accountNumber: P.select(P.string) }, value => ({
+          label: t("beneficiaries.accountIdentifier.accountNumber"),
+          text: value,
+        }))
+        .with({ IBAN: P.select(P.string) }, value => ({
+          label: t("beneficiaries.accountIdentifier.iban"),
+          text: printFormat(value),
+        }))
+        .with({ customerReferenceNumber: P.select(P.string) }, value => ({
+          label: t("beneficiaries.accountIdentifier.customerReferenceNumber"),
+          text: value,
+        }))
+        .with({ clabe: P.select(P.string) }, value => ({
+          label: t("beneficiaries.accountIdentifier.clabe"),
+          text: value,
+        }))
+        .with({ interacAccount: P.select(P.string) }, value => ({
+          label: t("beneficiaries.accountIdentifier.interacAccount"),
+          text: value,
+        }))
+        .otherwise(() => ({
+          label: "",
+          text: "",
+        })),
+    )
+    .otherwise(() => ({
+      label: "",
+      text: "",
+    }));
```

Comment:
can we rather use an `Option` or a nullish value to represent this?

Reply:
Done